ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderState.java

```diff
@@ -787,6 +789,46 @@ private void replicateNewConf() {
     return lists;
   }
 
+  /**
+   * See the thesis section 6.2: A leader in Raft steps down
+   * if an election timeout elapses without a successful
+   * round of heartbeats to a majority of its cluster.
+   */
+  private void checkLeadership() {
+    // The initial value of lastRpcResponseTime in FollowerInfo is set by
+    // LeaderState::addSenders(), which is fake and used to trigger an
+    // immediate round of AppendEntries request. To ensure that leader
+    // is stable, just ignore the check if become leader soon.
+    if (server.getRole().getRoleElapsedTimeMs() < 3 * server.getMaxTimeoutMs()) {
+      return;
+    }
+
+    final List<RaftPeerId> activePeers = senders.stream()
+        .filter(sender -> sender.getFollower()
+                                .getLastRpcResponseTime()
+                                .elapsedTimeMs() <= server.getMaxTimeoutMs())
```

Comment:
Should we have a separate config for the timeout value? I think it might be better to have a separate config.

Reply:
I am afraid not. Since the criterial is "A leader in Raft steps down if an election timeout elapses without a successful round of heartbeats to a majority of its cluster"